packages/blocks/src/api/raw-handling/utils.js

```diff
@@ -275,8 +275,8 @@ function cleanNodeList( nodeList, doc, schema, inline ) {
 			cleanNodeList( node.childNodes, doc, schema, inline );
 
 			// For inline mode, insert a line break when unwrapping nodes that
-			// are not phrasing content.
-			if ( inline && ! isPhrasingContent( node ) && node.nextElementSibling ) {
+			// are not phrasing content and no comments.
+			if ( inline && node.nodeType !== COMMENT_NODE && ! isPhrasingContent( node ) && node.nextElementSibling ) {
```

Comment:
Can these comments not be stripped beforehand? I think we'll need to add a filter to `deepFilterHTML`, right after `specialCommentConverter`.

Reply:
Hm, I can take a stab at it, sure. However, that same new filter would also be required in [`filterInlineHTML`](https://github.com/WordPress/gutenberg/blob/8195c7447ff2350055793b6e01a71b7a5b07720f/packages/blocks/src/api/raw-handling/paste-handler.js#L46), right? This function, too, calls `removeInvalidHTML`.